test/unit-test/dawn/Optimizer/Passes/TestFieldAccessIntervals.cpp

```diff
@@ -0,0 +1,184 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Compiler/Options.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/SIR/SIRSerializer.h"
+#include "test/unit-test/dawn/Optimizer/Passes/TestEnvironment.h"
+#include <fstream>
+#include <gtest/gtest.h>
+#include <streambuf>
+
+using namespace dawn;
+
+namespace {
+
+class TestFieldAccessIntervals : public ::testing::Test {
+  std::unique_ptr<dawn::Options> compileOptions_;
+
+  dawn::DawnCompiler compiler_;
+
+protected:
+  TestFieldAccessIntervals() : compiler_(compileOptions_.get()) {}
+  virtual void SetUp() {}
+
+  std::vector<std::shared_ptr<Stencil>> loadTest(std::string sirFilename) {
+
+    std::string filename = TestEnvironment::path_ + "/" + sirFilename;
+    std::ifstream file(filename);
+    DAWN_ASSERT_MSG((file.good()), std::string("File " + filename + " does not exists").c_str());
+
+    std::string jsonstr((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+    std::shared_ptr<SIR> sir =
+        SIRSerializer::deserializeFromString(jsonstr, SIRSerializer::SK_Json);
+
+    std::unique_ptr<OptimizerContext> optimizer = compiler_.runOptimizer(sir);
+    // Report diganostics
+    if(compiler_.getDiagnostics().hasDiags()) {
+      for(const auto& diag : compiler_.getDiagnostics().getQueue())
+        std::cerr << "Compilation Error " << diag->getMessage() << std::endl;
+      throw std::runtime_error("compilation failed");
+    }
+
+    DAWN_ASSERT_MSG((optimizer->getStencilInstantiationMap().count("compute_extent_test_stencil")),
+                    "compute_extent_test_stencil not found in sir");
+
+    return optimizer->getStencilInstantiationMap()["compute_extent_test_stencil"]->getStencils();
+  }
+};
+
+TEST_F(TestFieldAccessIntervals, test_field_access_interval_01) {
+  auto stencils = loadTest("test_field_access_interval_01.sir");
+  ASSERT_TRUE((stencils.size() == 1));
+  std::shared_ptr<Stencil> stencil = stencils[0];
+
+  ASSERT_TRUE((stencil->getNumStages() == 2));
```

Comment:
why is the extent calculation still here? Don't we do this in TestPAssComputeStageExtents?

Reply:
yes, but the input data set is different. Just to double check there is nothing weird?

Reply:
We can keep it if we assume that the serialized files might be corrupted.  It is an extra layer of safety but I'm not sure if it belongs into a unittest?
PassComputeStageExtents checks that the extent-calculation is correct and ../../SIR/TestSIRSerializer checks that the de-/ serialization works.